src/container/container.ts

```diff
@@ -408,11 +408,14 @@ class Container implements interfaces.Container {
     }
     private _removeModuleHandlers(moduleId:number): void {
         const handlers = this._moduleActivationStore.remove(moduleId)
-        if (handlers.onActivations.length > 0) {
-            this._activations.removeByCondition(onActivation => handlers.onActivations.indexOf(onActivation) !== -1);
-        }
-        if (handlers.onDeactivations.length > 0) {
-            this._deactivations.removeByCondition(onDeactivation => handlers.onDeactivations.indexOf(onDeactivation) !== -1);
+        this._removeModuleHandlersOfType(handlers,true);
+        this._removeModuleHandlersOfType(handlers,false);
+    }
+
+    private _removeModuleHandlersOfType(handlers:interfaces.ModuleActivationHandlers,activations:boolean): void {
+        const moduleHandlers = handlers[activations ? "onActivations":"onDeactivations"];
+        if(moduleHandlers.length > 0) {
+            this[activations ? "_activations":"_deactivations"].removeByCondition(handler => moduleHandlers.indexOf(handler as any) !== -1);
         }
```

Comment:
We should not refactor this. It's a false positive. Even if code climate suggests a refactor there is no way to improve the code

Reply:
index signatures are always hard to type properly. a ts PR is happening where you can you use a combined index signature based on symbols or even keys of enums. just keep those things in mind, especially me and @notaphplover since we tried to move to stricter rules and eslint.

Reply:
typecasting this things won't do much in this kind of cases since we probably need to cast to unknown first

Reply:
If you are not happy with some of these refactors can I leave it for you to revert them ?

Reply:
Sure, allow me to handle it :). It's still a great job @tonyhallett, thank you so much!

Reply:
Ok thanks.